cpp/src/ray/config_internal.cc

```diff
@@ -65,6 +67,9 @@ void ConfigInternal::Init(RayConfig &config, int *argc, char ***argv) {
     if (!FLAGS_ray_plasma_store_socket_name.empty()) {
       plasma_store_socket_name = FLAGS_ray_plasma_store_socket_name;
     }
+    if (!FLAGS_ray_session_dir.empty()) {
+      session_dir = FLAGS_ray_session_dir;
+    }
```

Comment:
```suggestion
    session_dir = FLAGS_ray_session_dir;
```

Reply:
Empty string is an invalid dir. My thought is that if FLAGS_ray_session_dir is empty, we can use default config which from hard code or default value.